pkg/controller/vpcpeering/peering.go

```diff
@@ -393,37 +375,111 @@ func (e *external) Update(ctx context.Context, mg cpresource.Managed) (managed.E
 	if !ok {
 		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
 	}
+	_, routeTableReady := cr.GetAnnotations()[routeTableEnsured]
+	_, hostZoneReady := cr.GetAnnotations()[hostedZoneEnsured]
+	_, attributeReady := cr.GetAnnotations()[attributeModified]
 
-	modifyVpcPeeringConnectionOptionsInput := &ec2.ModifyVpcPeeringConnectionOptionsInput{
-		VpcPeeringConnectionId: aws.String(meta.GetExternalName(cr)),
-		RequesterPeeringConnectionOptions: &ec2.PeeringConnectionOptionsRequest{
-			AllowDnsResolutionFromRemoteVpc: aws.Bool(true),
-		},
+	if !attributeReady {
+		modifyVpcPeeringConnectionOptionsInput := &ec2.ModifyVpcPeeringConnectionOptionsInput{
+			VpcPeeringConnectionId: cr.Status.AtProvider.VPCPeeringConnectionID,
+			RequesterPeeringConnectionOptions: &ec2.PeeringConnectionOptionsRequest{
+				AllowDnsResolutionFromRemoteVpc: aws.Bool(true),
+			},
+		}
+		_, err := e.client.ModifyVpcPeeringConnectionOptionsRequest(modifyVpcPeeringConnectionOptionsInput).Send(ctx)
+		if err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, errModifyVpcPeering)
+		}
+		if cr.Annotations == nil {
+			cr.Annotations = map[string]string{}
+		}
+		cr.Annotations[attributeModified] = "true"
+		err = e.kube.Update(ctx, cr)
+		if err != nil {
+			return managed.ExternalUpdate{}, awsclient.Wrap(err, "error update peering annotations")
+		}
 	}
 
-	_, err := e.client.ModifyVpcPeeringConnectionOptionsRequest(modifyVpcPeeringConnectionOptionsInput).Send(ctx)
-	if err != nil {
-		return managed.ExternalUpdate{}, errors.New(errModifyVpcPeering)
-	}
+	if !routeTableReady {
+		filter := ec2.Filter{
+			Name: aws.String("vpc-id"),
+			Values: []string{
+				*cr.Spec.ForProvider.VPCID,
+			},
+		}
+		describeRouteTablesInput := &ec2.DescribeRouteTablesInput{
+			Filters:    []ec2.Filter{filter},
+			MaxResults: aws.Int64(10),
+		}
+		routeTablesRes, err := e.client.DescribeRouteTablesRequest(describeRouteTablesInput).Send(ctx)
+		if err != nil {
+			return managed.ExternalUpdate{}, awsclient.Wrap(err, errDescribeRouteTable)
+		}
 
-	vpcAssociationAuthorizationInput := &route53.CreateVPCAssociationAuthorizationInput{
-		HostedZoneId: cr.Spec.ForProvider.HostZoneID,
-		VPC: &route53.VPC{
-			VPCId:     cr.Spec.ForProvider.PeerVPCID,
-			VPCRegion: route53.VPCRegion(*cr.Spec.ForProvider.PeerRegion),
-		},
+		for _, rt := range routeTablesRes.RouteTables {
+			createRouteInput := &ec2.CreateRouteInput{
+				RouteTableId:           rt.RouteTableId,
+				DestinationCidrBlock:   cr.Spec.ForProvider.PeerCIDR,
+				VpcPeeringConnectionId: cr.Status.AtProvider.VPCPeeringConnectionID,
+			}
+			createRouteRes, err := e.client.CreateRouteRequest(createRouteInput).Send(ctx)
+			if err != nil {
+				if aerr, ok := err.(awserr.Error); ok {
+					if aerr.Code() != "RouteAlreadyExists" {
```

Comment:
Use the `isAlreadyCreated` function?

Reply:
This code block is simply moved from the `Create()` function, `"RouteAlreadyExists"` is more specific than the `contains 'already'` criteria in the `isAlreadyCreated` function